pylib/mozhginfo/mozhginfo/mozhginfo.py

```diff
@@ -0,0 +1,167 @@
+#! /usr/bin/env python
+# coding: utf-8
+"""
+This helps us query information about Mozilla's Mercurial repositories.
+
+Documentation found in here:
+http://mozilla-version-control-tools.readthedocs.org/en/latest/hgmo/pushlog.html
+
+Important notes from the pushlog documentation::
+
+    When implementing agents that consume pushlog data, please keep in mind
+    the following best practices:
+
+    * Query by push ID, not by changeset or date.
+    * Always specify a startID and endID.
+    * Try to avoid full if possible.
+    * Always use the latest format version.
+    * Don't be afraid to ask for a new pushlog feature to make your life easier.
+"""
+import logging
+
+import requests
+
+
+LOG = logging.getLogger('pushlog_client')
+JSON_PUSHES = "%(repo_url)s/json-pushes"
+VALID_CACHE = {}
+
+
+class PushlogAuthError(Exception):
```

Comment:
Why an "Auth" error?
Pushlog is a public API which requires no auth.
e.g. https://hg.mozilla.org/try/json-pushes?changeset=71e69424094d&tipsonly=1&version=2&full=1